### PR TITLE
Fix random bug where regex doesn’t match every reg. expression

### PR DESCRIPTION
**Pull request by christophby**
Im my case it skipped random matches. This PR fixes it.

### CODE REVIEW

**Review by scottmas — APPROVED** [review]
I too faced this same annoying error and came upon the same fix, since RegExp.exec mutates the RegExp object

**Comment by scottmas** [comment]
In the meanwhile, while people are waiting for a fix, you can simply always ensure the pattern is a fresh instance in your render method. 
```javascript
const HASHTAG_REGEX = /#[a-z]+\b/i
...
pattern: new RegExp(HASHTAG_REGEX)
```

File: src/lib/TextExtraction.js [hunk 1]
```diff
@@ -31,7 +31,7 @@ class TextExtraction {
         let textLeft = parsedText.children;
 
         while (textLeft) {
-          let matches = pattern.pattern.exec(textLeft);
+          let matches = new RegExp(pattern.pattern).exec(textLeft);
```
Comment (scottmas) [inline src/lib/TextExtraction.js]: Would be _slightly_ more efficient to only clone it once outside the while loop...

**Comment by fbartho** [comment]
@christophby & @scottmas I'd be happy to ship this if the suggested change was made, and if we add tests that exercise this bug, so we make sure not to have a regression.

Please merge in master before you do that as I've switched the test harness to jest. 🥂

**Review by fbartho — CHANGES_REQUESTED** [review]
Please apply: 
https://github.com/taskrabbit/react-native-parsed-text/pull/47/files#r210374952

And please add tests!

**Review by shramadani — APPROVED** [review]

**Comment by shramadani** [comment]
> Please apply:
> https://github.com/taskrabbit/react-native-parsed-text/pull/47/files#r210374952
> 
> And please add tests!

Just tried this change locally and seems to be working as expected. Is it possible to merge this PR please?

**Comment by fbartho** [comment]
@shramadani -- if you want to help contribute tests for this PR, I'm looking to ship a new release of this this week. Please let me know if you're interested!

**Comment by ouoru** [comment]
@fbartho any updates on this? 😃

**Comment by fbartho** [comment]
Since there are no current tests, I'm going to try to write some myself. Additionally, there was a comment that I requested get applied, and that was resolved instead of applying it.

I probably will reset the regexp by setting `.lastIndex = 0` before the while loop assuming that doesn't break things.

This means I'm probably not going to merge this PR as is. I hope you understand @christophby

**Comment by fbartho** [comment]
There is an alternative implementation in this commit: https://github.com/taskrabbit/react-native-parsed-text/commit/e8f140e170a71c85e4358dcd9f64d593c10ad9fd

I will spin a beta build shortly

**Comment by fbartho** [comment]
`react-native-parsed-text@0.0.22-beta.1` is now available on npm -- please let me know if you have any issues!

**Comment by fbartho** [comment]
Closing this PR as there would now be conflicts, but please report if you have any issues with the current implementation, or if further fixes are necessary!

**Pull request closed without merging**